tests/tests/test-xcm/test-mock-dmp.ts

```diff
@@ -78,3 +79,153 @@ describeDevMoonbeam("Mock XCM - receive downward transfer", (context) => {
     expect(alith_dot_balance).to.eq(10n * RELAY_TOKEN);
   });
 });
+
+describeDevMoonbeam("Mock XCMP - test XCMP execution", (context) => {
+  it("Should test DMP on_initialization and on_idle", async function () {
+    const metadata = await context.polkadotApi.rpc.state.getMetadata();
+    const balancesPalletIndex = (metadata.asLatest.toHuman().pallets as Array<any>).find(
+      (pallet) => {
+        return pallet.name === "Balances";
+      }
+    ).index;
+    let ownParaId = (await context.polkadotApi.query.parachainInfo.parachainId()) as any;
+
+    let numMsgs = 50;
+    // let's target half of then being executed
+
+    // xcmp reserved is BLOCK/4
+    let totalDmpWeight =
+      context.polkadotApi.consts.system.blockWeights.maxBlock.toBigInt() / BigInt(4);
+
+    // we want half of numParaMsgs to be executed. That give us how much each message weights
+    let weightPerMessage = (totalDmpWeight * BigInt(2)) / BigInt(numMsgs);
+
+    let weightPerXcmInst = 100_000_000n;
+    // Now we need to construct the message. This needs to:
+    // - pass barrier (withdraw + clearOrigin*n buyExecution)
+    // - fail in buyExecution, so that the previous instruction weights are counted
+    // we know at least 2 instructions are needed per message (withdrawAsset + buyExecution)
+    // how many clearOrigins do we need to append?
+
+    // we will bias this number. The reason is we want to test the decay, and therefore we need
+    // an unbalanced number of messages executed. We specifically need that at some point
+    // we get out of the loop of the execution (we reach the threshold limit), to then
+    // go on idle
+
+    // for that reason, we multiply times 2
+    let clearOriginsPerMessage = (weightPerMessage - weightPerXcmInst * 2n) / weightPerXcmInst;
+
+    let instructions = [];
+    instructions.push({
+      WithdrawAsset: [
+        {
+          id: {
+            Concrete: {
+              parents: 0,
+              interior: {
+                X1: { PalletInstance: balancesPalletIndex },
+              },
+            },
+          },
+          fun: { Fungible: 1 },
+        },
+      ],
+    });
+
+    // we push clear origins
+    for (let i = 0; i < clearOriginsPerMessage; i++) {
+      instructions.push({
+        ClearOrigin: null,
+      });
+    }
```

Comment:
Entirely optional shorthand using `Array(count).fill(value)` syntax :P 

```suggestion
    instructions.push(...Array(clearOriginsPerMessage).fill({
      ClearOrigin: null,
    }});
```

Reply:
```
const instructions = [
    {
      WithdrawAsset: [
        {
          id: {
            Concrete: {
              parents: 0,
              interior: {
                X1: { PalletInstance: balancesPalletIndex },
              },
            },
          },
          fun: { Fungible: 1 },
        },
      ],
    },
    ...Array(clearOriginsPerMessage).fill({
	      ClearOrigin: null,
	  }),
	  {
      BuyExecution: {
        fees: {
          id: {
            Concrete: {
              parents: 1,
              interior: {
                X2: [{ Parachain: ownParaId }, { PalletInstance: balancesPalletIndex }],
              },
            },
          },
          fun: { Fungible: 1 },
        },
        weightLimit: { Limited: new BN(20000000000) },
      },
    }
];
```

Reply:
I have to do it like

```
...Array<XcmV1>(clearOriginsPerMessage).fill({
          ClearOrigin: null,
      }),
```

Reply:
Alright got it working with 

```
...Array(Number(clearOriginsPerMessage)).fill({
          ClearOrigin: null,
      }),
```